ViewModel/BrowserViewModel.swift

```diff
@@ -391,4 +392,34 @@ class BrowserViewModel: NSObject, ObservableObject,
             }
         }
     }
+
+    // MARK: - New tab in macos desktop
+      
+      #if os(macOS)
+      func webView(_ webView: WKWebView, createWebViewWith configuration: WKWebViewConfiguration,
+                   for navigationAction: WKNavigationAction, windowFeatures: WKWindowFeatures) -> WKWebView? {
+          
+          guard navigationAction.targetFrame == nil else { return nil }
+          guard let newUrl = navigationAction.request.url else { return nil }
+          
+          // open external link in default browser
+          if newUrl.isExternalLink {
+              self.openExternalLinkURL = newUrl
+              return nil
+          }
+
+          // create new tab
+          guard let currentWindow = NSApp.keyWindow,
+                let windowController = currentWindow.windowController else { return nil }
+          windowController.newWindowForTab(nil)
+          guard let newWindow = NSApp.keyWindow, currentWindow != newWindow else { return nil }
+          currentWindow.addTabbedWindow(newWindow, ordered: .above)
+          
+          // open url in new tab
+          NotificationCenter.openURL(newUrl, inNewTab: true)
```

Comment:
This works well most of the times, but not always. The issue I think is `@Environment(\.controlActiveState) var controlActiveState` is supposed to tell us which window is key, and only the key window should respond to the open URL notification, but there sometimes can be multiple key windows.

https://github.com/kiwix/apple/blob/fa397a905f95b946e1cf6b55328771b101bb91f8/App/App_macOS.swift#L141-L143

I am not an AppKit expert, but I was under the impression only one window can become key at a time, since only one window is supposed to be able to receive keyboard event

You can reproduce the issue with (on macOS 14):
- using right click, open in new window to open stay 4 tabs
- go to the 2nd tab, right click on something, and open in new window
- multiple windows responds to the notification 
  - with the additional problem that the newly opened window having the incorrect title (haven't investigated why)

<img width="1101" alt="Screenshot 2023-10-07 at 10 06 27 AM" src="https://github.com/kiwix/apple/assets/8294252/636fb982-b3e2-4a9f-9d72-b4e3b347c3cf">